lib/tidewave/database_adapter.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Tidewave
+  class DatabaseAdapter
+    class << self
+      def current
+        @current ||= create_adapter
+      end
+
+      def create_adapter
+        orm_type = detect_orm
+        case orm_type
+        when :active_record
+          require_relative "database_adapters/active_record"
+          DatabaseAdapters::ActiveRecord.new
+        when :sequel
+          require_relative "database_adapters/sequel"
+          DatabaseAdapters::Sequel.new
+        else
+          raise "No supported ORM detected. Please ensure ActiveRecord or Sequel is available."
+        end
+      end
+
+      private
+
+      def detect_orm
+        # Check for preferred ORM setting first
+        if defined?(Rails) && Rails.respond_to?(:application) && Rails.application.config.respond_to?(:tidewave)
```

Comment:
I am not sure we need to check for all of these contents, we assume in many other places we are running inside `Rails`.